doc/examples/applications/plot_3d_image_processing.py

```diff
@@ -0,0 +1,311 @@
+
+#####################################################################
+# As hinted before, a three-dimensional image can be viewed as a series of
+# two-dimensional planes. Let us write a helper function, `display`, to
+# display 30 planes of our data. By default, every other plane is displayed.
+
+
+def display(im3d, cmap="gray", step=2):
+    _, axes = plt.subplots(nrows=5, ncols=6, figsize=(16, 14))
+
+    vmin = im3d.min()
+    vmax = im3d.max()
+
+    for ax, image in zip(axes.flatten(), im3d[::step]):
+        ax.imshow(image, cmap=cmap, vmin=vmin, vmax=vmax)
+        ax.set_xticks([])
+        ax.set_yticks([])
+
+
+display(data)
+
+#####################################################################
+# Alternatively, we can explore these planes (slices) interactively using
+# Jupyter widgets. Let the user select which slice to display and show the
+# position of this slice in the 3D dataset.
+# Note that you cannot see the Jupyter widget at work in a static HTML page,
+# as is the case in the scikit-image gallery. For the following piece of
+# code to work, you need a Jupyter kernel running either locally or in the
+# cloud: see the bottom of this page to either download the Jupyter notebook
+# and run it on your computer, or open it directly in Binder.
+
+
+def slice_in_3D(ax, i):
+    # From https://stackoverflow.com/questions/44881885/python-draw-3d-cube
+    Z = np.array([[0, 0, 0],
+                  [1, 0, 0],
+                  [1, 1, 0],
+                  [0, 1, 0],
+                  [0, 0, 1],
+                  [1, 0, 1],
+                  [1, 1, 1],
+                  [0, 1, 1]])
+
+    Z = Z * data.shape
+    r = [-1, 1]
+    X, Y = np.meshgrid(r, r)
+
+    # Plot vertices
+    ax.scatter3D(Z[:, 0], Z[:, 1], Z[:, 2])
+
+    # List sides' polygons of figure
+    verts = [[Z[0], Z[1], Z[2], Z[3]],
+             [Z[4], Z[5], Z[6], Z[7]],
+             [Z[0], Z[1], Z[5], Z[4]],
+             [Z[2], Z[3], Z[7], Z[6]],
+             [Z[1], Z[2], Z[6], Z[5]],
+             [Z[4], Z[7], Z[3], Z[0]],
+             [Z[2], Z[3], Z[7], Z[6]]]
+
+    # Plot sides
+    ax.add_collection3d(
+        Poly3DCollection(
+            verts,
+            facecolors=(0, 1, 1, 0.25),
+            linewidths=1,
+            edgecolors="darkblue"
+        )
+    )
+
+    verts = np.array([[[0, 0, 0],
+                       [0, 0, 1],
+                       [0, 1, 1],
+                       [0, 1, 0]]])
+    verts = verts * (60, 256, 256)
+    verts += [i, 0, 0]
+
+    ax.add_collection3d(
+        Poly3DCollection(
+            verts,
+            facecolors="magenta",
+            linewidths=1,
+            edgecolors="black"
+        )
+    )
+
+    ax.set_xlabel("plane")
+    ax.set_xlim(0, 100)
+    ax.set_ylabel("row")
+    ax.set_zlabel("col")
+
+    # Autoscale plot axes
+    scaling = np.array([getattr(ax,
+                                "get_{}lim".format(dim))() for dim in "xyz"])
```

Comment:
```suggestion
                                f'get_{dim}lim')() for dim in "xyz"])
```

Reply:
@jni style question: I learnt Python using single quotes by default, resorting to double quotes only when necessary. Then `black` influenced me into double quotes... Do you have a strong leaning?

Reply:
Yes, my strong leaning is that [black is the devil](https://github.com/napari/napari/pull/1451#issuecomment-658002959), and your history reinforces that opinion. =P

More seriously: I feel very strongly that the original PEP8 had it right (single quotes by default), and black really made the wrong decision here, as with their default line length. They essentially had to admit as such, adding config options for both of these in a package generally devoid of config options.

In scikit-image, when in conflict, follow PEP8, not black.

I *will* also say that we have not actually enforced string formatting at all. I only changed these because I was changing them to f-strings anyway. I wouldn't stress about going back and changing existing stuff.

Reply:
Ok, thanks for sharing! =)

Yes, yes, I got that the suggested change was about f-strings and not so much about quotes. But this side 'issue' had been on my mind, so I'm happy to be more informed now. Thanks.